File: app/src/main/java/com/demo/adapter/MainActivity.java
package com.demo.adapter;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.demo.adapter.model.tab.StickyActivity2;

import java.util.Arrays;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;
import lib.kalu.adapter.BaseCommonAdapter;
import lib.kalu.adapter.holder.RecyclerHolder;
import lib.kalu.adapter.manager.CrashGridLayoutManager;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final List<String> list = Arrays.asList("点击事件", "加载更多", "头部尾部", "多种布局", "空布局", "悬浮菜单", "TabMore", "Tab", "分组", "侧滑拖动", "分组折叠");
        final List<Class<? extends AppCompatActivity>> clazzs = Arrays.asList(ClickActivity.class, LoadmoreActivity.class, HeadFootActivity.class, MulitActivity.class, EmptyActivity.class, FloatActivity.class, StickyActivity2.class, StickyActivity1.class, SectionActivity.class, DragSwipeActivity.class, TransActivity.class);

        BaseCommonAdapter<String> adapter = new BaseCommonAdapter<String>() {

            @Override
            protected int onView() {
                return R.layout.activity_main_item;
            }

            @NonNull
            @Override
            protected List<String> onData() {
                return list;
            }

            @Override
            protected void onNext(RecyclerHolder holder, String model, final int position) {

                holder.setText(R.id.main_text, model);
                holder.getView(R.id.main_text).setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        startActivity(new Intent(getApplicationContext(), clazzs.get(position)));
                    }
                });
            }
        };

        RecyclerView view = findViewById(R.id.list);
        view.setLayoutManager(new CrashGridLayoutManager(getApplicationContext(), 2));
        view.setAdapter(adapter);
    }
}

File: app/src/main/java/com/demo/adapter/MainActivity.kt
package com.demo.adapter

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView
import android.view.View
import com.demo.adapter.model.tab.StickyActivity2
import lib.kalu.adapter.BaseCommonAdapter
import lib.kalu.adapter.decoration.SpaceDecoration
import lib.kalu.adapter.holder.RecyclerHolder
import lib.kalu.adapter.manager.CrashGridLayoutManager
import lib.kalu.adapter.manager.CrashLinearLayoutManager
import java.util.*

class MainActivity : AppCompatActivity() {

    private val mActivity: Array<Class<*>>? by lazy {

        arrayOf<Class<*>>(ClickActivity::class.java, LoadmoreActivity::class.java, HeadFootActivity::class.java,
                MulitActivity::class.java, EmptyActivity::class.java,
                FloatActivity::class.java,
                StickyActivity2::class.java, StickyActivity1::class.java,
                SectionActivity::class.java,
                DragSwipeActivity::class.java,
                TransActivity::class.java)
    }

    private val mTitle: Array<String>? by lazy {
        arrayOf("点击事件", "加载更多", "头部尾部",
                "多种布局", "空布局",
                "悬浮菜单",
                "TabMore", "Tab",
                "分组",
                "侧滑拖动",
                "分组折叠")
    }

    private val mRecyclerView: RecyclerView? by lazy {
        findViewById(R.id.main_recycler) as RecyclerView
    }

    private val mArrayList: ArrayList<String>? by lazy {
        ArrayList<String>()
    }

    private val mLayoutManager: CrashGridLayoutManager? by lazy {
        CrashGridLayoutManager(applicationContext, 2)
    }

    private val mSpaceDecoration: SpaceDecoration? by lazy {
        SpaceDecoration(10)
    }

    private val mLoadAdapter: BaseCommonAdapter<String> by lazy {

        object : BaseCommonAdapter<String>(mArrayList, R.layout.activity_main_item) {

            override fun onNext(holder: RecyclerHolder, model: String, position: Int) {

                holder!!.setText(R.id.main_text, model)
                holder!!.setOnClickListener(View.OnClickListener {

                    startActivity(Intent(applicationContext, mActivity!![position]))
                }, R.id.main_text)
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        for (i in 0 until mTitle!!.size) {
            mArrayList?.add(mTitle!![i])
        }

        mRecyclerView!!.addItemDecoration(mSpaceDecoration)
        mRecyclerView!!.layoutManager = mLayoutManager
        mRecyclerView!!.adapter = mLoadAdapter
    }
}